diff -Naur crazycat-2017-11-13.orig/linux/Makefile crazycat-2017-11-13/linux/Makefile
--- crazycat-2017-11-13.orig/linux/Makefile	2017-12-03 20:47:44.000000000 +0100
+++ crazycat-2017-11-13/linux/Makefile	2017-12-20 09:48:15.516826488 +0100
@@ -112,6 +112,13 @@
 
 untar: linux-media.tar.bz2
 	tar xfj linux-media.tar.bz2
+	mkdir -p drivers/media/amlogic/
+	# Copy amlvideodri module
+	cp -a "$(SRCDIR)/drivers/amlogic/video_dev" "drivers/media/amlogic"
+	sed -i 's,common/,,g; s,"trace/,",g' `find drivers/media/amlogic/video_dev/ -type f`
+	# Copy videobuf-res module
+	cp -a "$(SRCDIR)/drivers/media/v4l2-core/videobuf-res.c" "drivers/media/v4l2-core/"
+	cp -a "$(SRCDIR)/include/media/videobuf-res.h" "include/media/"
 	-rm -f .patches_applied .linked_dir .git_log.md5
 
 clean:
