diff --git a/src/main.cpp b/src/main.cpp
index a93a109..a21f445 100755
--- a/src/main.cpp
+++ b/src/main.cpp
@@ -1480,7 +1480,9 @@ void Session::GetSupportedDecrypterURN(std::string &key_system)
 #ifdef ANDROID
   searchPaths[0] = kodi::vfs::TranslateSpecialProtocol("special://xbmcbinaddons/");
 #else
-  searchPaths[0] = kodi::vfs::TranslateSpecialProtocol("special://xbmcbinaddons/inputstream.adaptive/");
+  searchPaths[0] = kodi::vfs::TranslateSpecialProtocol(specialpath);
+  kodihost.SetLibraryPath(searchPaths[0].c_str());
+  //searchPaths[0] = kodi::vfs::TranslateSpecialProtocol("special://xbmcbinaddons/inputstream.adaptive/");
 #endif
   searchPaths[1] = kodi::GetAddonInfo("path");
