diff -Naur libcec-5250931.orig/src/libcec/implementations/SLCommandHandler.cpp libcec-5250931/src/libcec/implementations/SLCommandHandler.cpp
--- libcec-5250931.orig/src/libcec/implementations/SLCommandHandler.cpp	2017-01-23 16:37:10.000000000 +0100
+++ libcec-5250931/src/libcec/implementations/SLCommandHandler.cpp	2017-12-11 11:00:50.541261538 +0100
@@ -129,7 +129,7 @@
   else if (command.parameters.size == 1 &&
       command.parameters[0] == SL_COMMAND_REQUEST_RECONNECT)
   {
-    HandleVendorCommandPowerOn(command);
+    HandleVendorCommandPowerOnStatus(command);
     return COMMAND_HANDLED;
   }
   else if (command.parameters.size == 1 &&
