commit 79aa2b90e8d9ff5872e0bd8adf121d8b4b500131
Author: Lukas Rusak <lorusak@gmail.com>
Date:   Mon Dec 11 23:44:41 2017 -0800

    [linux] allow using alsa and pulse together

diff --git a/xbmc/windowing/X11/WinSystemX11GLContext.cpp b/xbmc/windowing/X11/WinSystemX11GLContext.cpp
index 72a70d8118..0914d28820 100644
--- a/xbmc/windowing/X11/WinSystemX11GLContext.cpp
+++ b/xbmc/windowing/X11/WinSystemX11GLContext.cpp
@@ -52,24 +52,8 @@ std::unique_ptr<CWinSystemBase> CWinSystemBase::CreateWinSystem()
 
 CWinSystemX11GLContext::CWinSystemX11GLContext()
 {
-  std::string envSink;
-  if (getenv("AE_SINK"))
-    envSink = getenv("AE_SINK");
-  if (StringUtils::EqualsNoCase(envSink, "ALSA"))
-  {
-    X11::ALSARegister();
-  }
-  else if (StringUtils::EqualsNoCase(envSink, "PULSE"))
-  {
-    X11::PulseAudioRegister();
-  }
-  else
-  {
-    if (!X11::PulseAudioRegister())
-    {
-      X11::ALSARegister();
-    }
-  }
+  X11::ALSARegister();
+  X11::PulseAudioRegister();
 }
 
 CWinSystemX11GLContext::~CWinSystemX11GLContext()
diff --git a/xbmc/windowing/amlogic/WinSystemAmlogic.cpp b/xbmc/windowing/amlogic/WinSystemAmlogic.cpp
index 4b8199760e..e50a31daf9 100644
--- a/xbmc/windowing/amlogic/WinSystemAmlogic.cpp
+++ b/xbmc/windowing/amlogic/WinSystemAmlogic.cpp
@@ -32,6 +32,7 @@
 // AESink Factory
 #include "cores/AudioEngine/AESinkFactory.h"
 #include "cores/AudioEngine/Sinks/AESinkALSA.h"
+#include "cores/AudioEngine/Sinks/AESinkPULSE.h"
 #include "guilib/GraphicContext.h"
 #include "guilib/Resolution.h"
 #include "settings/Settings.h"
@@ -78,6 +79,7 @@ CWinSystemAmlogic::CWinSystemAmlogic()
   // Register sink
   AE::CAESinkFactory::ClearSinks();
   CAESinkALSA::Register();
+  CAESinkPULSE::Register();
 }
 
 CWinSystemAmlogic::~CWinSystemAmlogic()
diff --git a/xbmc/windowing/gbm/WinSystemGbm.cpp b/xbmc/windowing/gbm/WinSystemGbm.cpp
index b3daea6569..0cd93b4566 100644
--- a/xbmc/windowing/gbm/WinSystemGbm.cpp
+++ b/xbmc/windowing/gbm/WinSystemGbm.cpp
@@ -33,24 +33,8 @@ CWinSystemGbm::CWinSystemGbm() :
   m_nativeDisplay(nullptr),
   m_nativeWindow(nullptr)
 {
-  std::string envSink;
-  if (getenv("AE_SINK"))
-    envSink = getenv("AE_SINK");
-  if (StringUtils::EqualsNoCase(envSink, "ALSA"))
-  {
-    GBM::ALSARegister();
-  }
-  else if (StringUtils::EqualsNoCase(envSink, "PULSE"))
-  {
-   GBM::PulseAudioRegister();
-  }
-  else
-  {
-    if (!GBM::PulseAudioRegister())
-    {
-      GBM::ALSARegister();
-    }
-  }
+  GBM::ALSARegister();
+  GBM::PulseAudioRegister();
 
   m_winEvents.reset(new CWinEventsLinux());
 }
diff --git a/xbmc/windowing/rpi/WinSystemRpi.cpp b/xbmc/windowing/rpi/WinSystemRpi.cpp
index 3e4c6d2d8a..a00a1c0c18 100644
--- a/xbmc/windowing/rpi/WinSystemRpi.cpp
+++ b/xbmc/windowing/rpi/WinSystemRpi.cpp
@@ -34,7 +34,9 @@
 #include "utils/log.h"
 #include "../WinEventsLinux.h"
 #include "cores/AudioEngine/AESinkFactory.h"
+#include "cores/AudioEngine/Sinks/AESinkALSA.h"
 #include "cores/AudioEngine/Sinks/AESinkPi.h"
+#include "cores/AudioEngine/Sinks/AESinkPULSE.h"
 
 #include <EGL/egl.h>
 #include <EGL/eglplatform.h>
@@ -55,6 +57,8 @@ CWinSystemRpi::CWinSystemRpi()
   m_winEvents.reset(new CWinEventsLinux());
   AE::CAESinkFactory::ClearSinks();
   CAESinkPi::Register();
+  CAESinkALSA::Register();
+  CAESinkPULSE::Register();
 }
 
 CWinSystemRpi::~CWinSystemRpi()
